Library/Formula/telegram.rb

```diff
@@ -0,0 +1,40 @@
+class Telegram < Formula
+  homepage "https://github.com/vysheng/tg"
+  url "https://github.com/vysheng/tg/archive/1.0.5.1.tar.gz"
+  sha1 "72f6a9f89ee9db90b573bad25e567a2015115848"
+
+  depends_on "openssl"
+  depends_on "libevent"
+  depends_on "libconfig"
+  depends_on "readline"
+  depends_on "lua"
+
+  patch :DATA
+
+  def install
+    ENV.append "CFLAGS", "-I#{HOMEBREW_PREFIX}/include -I#{Formula["readline"].prefix}/include"
+    ENV.append "LDFLAGS", "-L#{HOMEBREW_PREFIX}/lib -L#{Formula["readline"].prefix}/lib"
+
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+
+    bin.install "bin/telegram-cli"
+    etc.install "tg-server.pub"
```

Comment:
There's a open pull request in upstream, but it isn't merged (the patch was submitted almost 2 years ago) :-(
https://github.com/vysheng/tg/pull/19